### PR TITLE
Map

### PR DESCRIPTION
**Pull request by QimatLuo**
countf(), all(), reject(), chunk() they are just pass callback to map(). So they need to migrate together.

### CODE REVIEW

File: moses.lua [hunk 1]
```diff
@@ -27,7 +27,7 @@ local function f_max(a,b) return a>b end
 local function f_min(a,b) return a<b end
 local function clamp(var,a,b) return (var<a) and a or (var>b and b or var) end
 local function isTrue(_,value) return value and true end
-local function iNot(value) return not value end
+local function iNot(_,value) return not value end
```
Comment (akiratsai) [inline moses.lua]: we don't need to change this line

Comment (akiratsai) [inline moses.lua]: we don't need to change this line

Comment (QimatLuo) [reply moses.lua]: because iNot() is using in _.reject() callback.

Comment (akiratsai) [reply moses.lua]: `_.reject` is using `_.map` and `_.map` is using `function(v, k)`

Comment (QimatLuo) [reply moses.lua]: Right, so originally _map() is using (k, v), now it's (v,k).
That's why I change it to iNot(_,value).

**Review by ghost — APPROVED** [review]

**Pull request merged**